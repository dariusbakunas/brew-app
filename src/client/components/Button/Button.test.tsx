import { shallow } from 'enzyme';
import React from 'react';
import { Button } from '../index';

describe('Button', () => {
  it('should render without throwing an error', () => {
    expect(shallow(<Button>test</Button>).contains(
      <button className='uk-button uk-button-default' type='button'>test</button>,
    )).toBeTruthy();
  });

  it('should call onClick handler on click event', () => {
    const handler = jest.fn();
    const wrapper = shallow(<Button onClick={handler}/>);
    wrapper.find('button').simulate('click');
    expect(handler).toHaveBeenCalledTimes(1);
  });

  it('should apply additional classes', () => {
    const wrapper = shallow(<Button className='test-class'/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders icon on the left by default', () => {
    const wrapper = shallow(<Button icon='trash'>test</Button>);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders icon on the right if specified', () => {
    const wrapper = shallow(<Button icon='trash' iconPosition='right'>test</Button>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should set variation class', () => {
    const wrapper = shallow(<Button variation='primary'>test</Button>);
    expect(wrapper).toMatchSnapshot();
  });
});
