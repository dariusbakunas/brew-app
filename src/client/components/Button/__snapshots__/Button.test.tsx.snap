// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders icon on the left by default 1`] = `
<button
  className="uk-button icon"
  type="button"
>
  <span
    className="uk-icon"
  >
    <Icon
      icon="trash"
      width={20}
    />
  </span>
   
  <span>
    test
  </span>
</button>
`;

exports[`Button renders icon on the right if specified 1`] = `
<button
  className="uk-button icon"
  type="button"
>
  <span>
    test
  </span>
  <span
    className="uk-icon"
  >
    <Icon
      icon="trash"
      width={20}
    />
  </span>
   
</button>
`;

exports[`Button should apply additional classes 1`] = `
<button
  className="uk-button uk-button-default test-class"
  type="button"
/>
`;

exports[`Button should set variation class 1`] = `
<button
  className="uk-button uk-button-primary"
  type="button"
>
  test
</button>
`;
